src/app/(proper_react)/redesign/(authenticated)/user/dashboard/Dashboard.test.tsx

```diff
@@ -75,9 +143,33 @@ it("shows the premium upgrade cta if the user is not a premium subscriber", () =
     name: "Upgrade to ⁨Premium⁩",
   });
   expect(premiumCtas.length).toBe(2);
+
+  await user.click(premiumCtas[0]);
+
+  const productTabYearly1 = screen.queryByRole("tab", { name: "Yearly" });
+  expect(productTabYearly1?.getAttribute("aria-selected")).toBe("true");
+  const productTabMonthly1 = screen.queryByRole("tab", { name: "Monthly" });
+  expect(productTabMonthly1?.getAttribute("aria-selected")).toBe("false");
```

Comment:
Awesome job checking for these `aria-` properties, to ensure no accessibility regressions as well :clap: